### PR TITLE
Add release and slow jobs for Trusty stable

### PR DESCRIPTION
**Pull request by wonderfly**
@ixdy

### CODE REVIEW

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -520,13 +526,13 @@ case ${JOB_NAME} in
     ;;
 
   # Runs slow tests on GCE with Trusy as base image for minions, sequentially.
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: typo: Trusy

File: hack/jenkins/e2e.sh [hunk 2]
```diff
@@ -622,6 +628,48 @@ case ${JOB_NAME} in
     : ${FAIL_ON_GCP_RESOURCE_LEAK:="true"}
     ;;
 
+  # Runs non-flaky tests on GCE with Trusty-stable as base image for minions,
```
Comment (ixdy) [inline hack/jenkins/e2e.sh]: should this comment say "Trusty-beta"?

Comment (ixdy) [inline hack/jenkins/e2e.sh]: oh, no, the job name is wrong.

File: hack/jenkins/e2e.sh (hunk 1 above)
Comment (wonderfly) [inline hack/jenkins/e2e.sh]: Done

File: hack/jenkins/e2e.sh [hunk 3]
```diff
@@ -560,7 +566,7 @@ case ${JOB_NAME} in
    : ${JENKINS_PUBLISHED_VERSION:="release/stable-1.1"}
    ;;
```
Comment (wonderfly) [inline hack/jenkins/e2e.sh]: Yeah, sorry about that. Updated.

**Comment by ixdy** [comment]
LGTM

**Comment by wonderfly** [comment]
What are we blocked on to get this merged?

**Comment by ixdy** [comment]
@k8s-bot e2e test this, issue #IGNORE

e2e tests are flaky on release-1.1. I'm not sure filing an issue about this is worth it.

**Comment by ixdy** [comment]
@k8s-bot e2e test this again, issue #13485
@k8s-bot unit test this again, issue #18809

**Pull request merged**